pkg/security/loaders/raiffeisench/raiffeisench.go

```diff
@@ -0,0 +1,156 @@
+package raiffeisench
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/enrichman/portfolio-perfomance/pkg/security"
+)
+
+type RaiffeisenchQuoteLoader struct {
+	name  string
+	isin  string
+	valor int
+}
+
+func New(name, isin string) *RaiffeisenchQuoteLoader {
+	valor, err := getValorFromISIN(isin)
+	if err != nil {
+		panic(err)
+	}
+
+	return &RaiffeisenchQuoteLoader{
+		name:  name,
+		isin:  isin,
+		valor: valor,
+	}
+}
+
+func getValorFromISIN(isin string) (int, error) {
+	// https://en.wikipedia.org/wiki/Valoren_number
+	// https://www.isin.org/isin-format/
+
+	if !strings.HasPrefix(isin, "CH") {
+		return 0, fmt.Errorf("valor number can only be extracted for Swiss financial instruments")
+	}
+
+	// strip CH
+	val := strings.TrimPrefix(isin, "CH")
+
+	// strip all leading zeros
+	val = strings.TrimLeft(val, "0")
+
+	// drop last digit (ISIN checksum)
+	val = val[0 : len(val)-1]
+
+	valor, err := strconv.Atoi(val)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read valor number from ISIN: %w", err)
+	}
+	return valor, nil
+}
+
+func (r *RaiffeisenchQuoteLoader) Name() string {
+	return r.name
+}
+
+func (r *RaiffeisenchQuoteLoader) ISIN() string {
+	return r.isin
+}
+
+func (r *RaiffeisenchQuoteLoader) Valor() int {
+	return r.valor
+}
+
+type HistoryQuotesRequest struct {
+	Valor      int       `json:"valor"`
+	ExchangeId int       `json:"exchangeId"`
+	CurrencyId int       `json:"currencyId"`
+	From       time.Time `json:"from"`
+	To         time.Time `json:"to"`
+}
+
+type HistoryQuotesResponse struct {
+	HistoryQuotes []struct {
+		Date  string      `json:"date"`
+		Close float64     `json:"close"`
+		Open  interface{} `json:"open"`
+		High  interface{} `json:"high"`
+		Low   interface{} `json:"low"`
+	} `json:"historyQuotes"`
+}
+
+func (r *RaiffeisenchQuoteLoader) LoadQuotes() ([]security.Quote, error) {
+	// Build request payload
+	endDate := time.Now()
+	startDate := time.Now().Add(-1 * time.Hour * 24 * 365) // one year ago
+	payload := HistoryQuotesRequest{
+		Valor:      r.valor,
+		ExchangeId: 3233,
+		CurrencyId: 1, // CHF
+		From:       startDate.UTC(),
+		To:         endDate.UTC(),
+	}
+
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling request body: %w", err)
+	}
+
+	// Build request
+	req, _ := http.NewRequest(
+		"POST",
+		"https://boerse.raiffeisen.ch/api/HistoryQuotes",
+		bytes.NewBuffer(payloadBytes),
+	)
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-type", "application/json")
+	// We need to directly set the key in the header map,
+	// because Go's stdlib canonicalizes all HTTP headers,
+	// but the remote API requires this header to be lowercase.
+	req.Header["customer"] = []string{"raiffeisen-prod"}
+
+	// fmt.Println(req)
```

Comment:
It would actually be nice if the loader implementation could get the LOG_LEVEL flag that is used in `main`, that way all this could be wired up to output when `LOG_LEVEL == debug`.
But I didn't want to touch the interface in this PR, so I left the commented out debugging statements for later :-)